Better handling of original timestamp

ph1lm opened the pull request:
*Description of changes:*
Predicates are added for point-in-time replication.
Also, an opportunity to replicate the original timestamp was added.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

ph1lm commented:
Hi @nwheeler81 
Sorry for delay 
I took some time to get it approved

nwheeler81 commented on glue/README.MD:
```diff
@@ -136,6 +136,20 @@ on the target table before running the following command
+                {
+                 "replication": {
+                   "replicateWithTimestamp": true
+                }'
```
missing `}`

ph1lm replied:
Fixed

nwheeler81 commented:
Tested use-cases:
Table structure (id text PRIMARY KEY, code timeuuid, value text):  
1. ```./cqlreplicator --state run --tiles 1 --landing-zone "s3://cql-replicator-account-us-east-1-pitr" --region us-east-1 --worker-type G.025X --writetime-column value --src-keyspace ks_pitr --src-table tbl_pitr --trg-keyspace ks_pitr --trg-table tbl_pitr --env issue151 --json-mapping '{ "replication": {"replicateWithTimestamp": true}}' ``` - **passed** 

2. ```./cqlreplicator --state run --tiles 1 --landing-zone "s3://cql-replicator-account-us-east-1-pitr" --region us-east-1 --worker-type G.025X --srf 1722270672200697 --writetime-column value --src-keyspace ks_pitr --src-table tbl_pitr --trg-keyspace ks_pitr --trg-table tbl_pitr --env issue151 --json-mapping ' {"replication": { "replicateWithTimestamp": true,"pointInTimeReplicationConfig": {"predicateOp": "lessThan"} }}'``` - **passed**

3. ```./cqlreplicator --state run --tiles 1 --landing-zone "s3://cql-replicator-account-us-east-1-pitr" --region us-east-1 --worker-type G.025X --srf 1722270672200697 --writetime-column value --src-keyspace ks_pitr --src-table tbl_pitr --trg-keyspace ks_pitr --trg-table tbl_pitr --env issue151 --json-mapping ' {"replication": { "replicateWithTimestamp": true,"pointInTimeReplicationConfig": {"predicateOp": "greaterThan"} }}'``` - **passed**

nwheeler81 commented:
@michaelraney could you please review #154

michaelraney commented:
Interested to know what is the use case for "lessThanOrEqual"

ph1lm commented:
@michaelraney We used it for our migration from C* to K*. 

We implemented dual-write capability in our application and enabled it in prod so it was writing to both C* and K*, but still reading from C*, and we captured a timestamp. So, all new data after this timestamp is replicated from C* to K* by dual-write.

After that, we launched the cql-replicator with lessThanOrEqual=timestamp option.
So, all old data before the timestamp is replicated from C* to K* by cql-replicator.

After cql-replicator finished, we disabled dual-write and switched references from C* to K* for both reads and writes.

michaelraney commented:
Great thank you,

michaelraney reviewed (approved):
Looks good
